File: c++/lec2_jan28/twoLines.cpp
#include <iostream>
using namespace std;

int main(int argc, char** argv){
	int x1,x2;
	int x3,x4;

	int d;

	cout << "Enter x1 and x2: \n";
	cin >> x1 >> x2;

	cout << "Enter x3 and x4: \n";
	cin >> x3 >> x4;

	if(x1 > x2){
		d = x2;
		x2 = x1;
		x1 = d; 
	}

	if(x3 > x4){
		d = x4;
		x4 = x3;
		x3 = d;
	}

	if(x1 >= x4 || x3 >= x2){
		cout << "No Overlap" << endl;
	} else if(x1 <= x3 && x2 >= x4){
		cout << "L2 is within L1" << endl;
	} else if(x3 <= x1 && x4 >= x2){
		cout << "L1 is within L2" << endl;
	} else {
		cout << "L1 and L2 are overlapping lines" << endl;
	}
}

File: rust/lec2_jan28/two_lines.rs
use std::io;

fn main() {
	let mut x1 = String::new();
	let mut x2 = String::new();
	let mut y1 = String::new();
	let mut y2 = String::new();

	io::stdin().read_line(&mut x1)
		.expect("Err");
	io::stdin().read_line(&mut x2)
		.expect("Err");
	io::stdin().read_line(&mut y1)
		.expect("Err");
	io::stdin().read_line(&mut y2)
		.expect("Err");

	let mut x1: i32 = x1.trim().parse()
		.expect("enter a coordinate");
	let mut x2: i32 = x2.trim().parse()
		.expect("enter a coordinate");
	let mut y1: i32 = y1.trim().parse()
		.expect("enter a coordinate");
	let mut y2: i32 = y2.trim().parse()
		.expect("enter a coordinate");

	if x1 > x2 {
		let temp: i32 = x2;
		x2 = x1;
		x1 = temp;
	}

	if y1 > y2 {
		let temp: i32 = y2;
		y2 = y1;
		y1 = temp;
	}

	if (x2 < y1) || (y2 < x1) {
		println!("No Overlap");
	} else if (y1 > x1 && y2 < x2) || (x1 > y1 && x2 < y2) {
		println!("L1 in L2 or L2 in L1");
	} else {
		println!("Intersecting lines");
	}
}